packages/sheets-ui/src/controllers/editor/formula-editor.controller.ts

```diff
@@ -157,6 +158,12 @@ export class FormulaEditorController extends RxDisposable {
         });
     }
 
+    private _listenFoldBtnClick() {
+        this._formulaEditorManagerService.foldBtnStatus$.pipe(takeUntil(this.dispose$)).subscribe(() => {
```

Comment:
What if there are other events that may change the workbench's layout? An abstraction of this type of events would be better.